app/code/community/Bolt/Boltpay/controllers/ApiController.php

```diff
@@ -277,8 +278,8 @@ public function create_orderAction() {
             $computedCart = Mage::getModel('boltpay/boltOrder')->buildCart($immutableQuote, false );
             $this->boltHelper()->notifyException($orderCreationException, array( 'magento_order_details' => json_encode($computedCart)));
             //////////////////////////////////////////////////////
-
-        } finally {
+            // finally requre PHP => 5.5
+            //} finally {
             benchmark( "Response sent to Bolt");
         }
```

Comment:
I think you have to re-throw from `catch`, otherwise this changes behavour? (previously exception was thrown after executing finally, now you're swallowing error)

Reply:
> I think you have to re-throw from catch, otherwise this changes behavour? (previously exception was thrown after executing finally, now you're swallowing error)

@daisy1754  and @oldDriver, **True**

Therefore, ideally, the code will end up looking like:
```
$finallyFunction = function() { benchmark( "Response sent to Bolt"); };
try {
     ... // same code as before
} catch ( Bolt_Boltpay_OrderCreationException $orderCreationException) {
     ...  // same code as before
} catch ( Exception $e) {
     // catch-all workaround for finally not being supported in PHP 5.4
      $this->boltHelper()->notifyException($orderCreationException, array( 'magento_order_details' => json_encode($computedCart)));
     $finallyFunction();
     throw $e;
}
$finallyFunction();
```